Mathlib/Algebra/BigOperators/Group/Finset.lean

```diff
@@ -1165,6 +1165,17 @@ theorem prod_ite_eq' [DecidableEq α] (s : Finset α) (a : α) (b : α → β) :
     (∏ x ∈ s, ite (x = a) (b x) 1) = ite (a ∈ s) (b a) 1 :=
   prod_dite_eq' s a fun x _ => b x
 
+@[to_additive]
+theorem prod_ite_eq_of_mem [DecidableEq α] (s : Finset α) (a : α) (b : α → β) (h : a ∈ s) :
+    (∏ x ∈ s, if a = x then b x else 1) = b a := by
+  simp only [prod_ite_eq, h, ↓reduceIte]
+
+/-- The difference with `Finset.prod_te_eq_of_mem` is that the arguments ot `Eq` are swapped. -/
```

Comment:
```suggestion
/-- The difference with `Finset.prod_ite_eq_of_mem` is that the arguments to `Eq` are swapped. -/
```

Reply:
I use `sum_ite_eq_of_mem'` five times in the draft PR. Admittedly I only rewrite in reverse once [here](https://github.com/leanprover-community/mathlib4/pull/20008/files#diff-0c922162d6078c849fff74f8322f93ac418fbfb2a413cfbe1204e744472df019R211), and indeed it needs some pretty strong unification hints. I can get rid of the reverse rewrite by adding an extra line to the calc block, which may be the right thing to do here. 

That said I just attempted to replace all uses of `rw [sum_ite_eq_of_mem']` with `rw [sum_ite_eq', if_pos]` and I ran into a usability issue [here](https://github.com/leanprover-community/mathlib4/pull/20008/files#diff-0c922162d6078c849fff74f8322f93ac418fbfb2a413cfbe1204e744472df019R308). What happens is that the sum I'm removing is on the RHS of the equation, and the following `if_pos` matches with an `if then else` block on the LHS. Perhaps this is enough of a reason to keep these lemmas?